substratevm/src/com.oracle.svm.driver/src/com/oracle/svm/driver/NativeImage.java

```diff
@@ -1204,6 +1205,18 @@ protected int buildImage(List<String> javaArgs, LinkedHashSet<Path> bcp, LinkedH
 
         int exitStatus = 1;
         try {
+            if (config.useJavaModules()) {
+                /* For Java > 8 we use an argument file to pass the options to the builder */
+                Path argsFile = Files.createTempFile("native-image", "args");
+                argsFile.toFile().deleteOnExit();
```

Comment:
Unconditionally deleting `argsFile` makes it harder to replay and debug failures. I would suggest only deleting it if `exitStatus` below is non-zero. In any case, no need to use `deleteOnExit` - just add a `finally` clause to the `try` statement.

Reply:
@dougxc see the comments @olpaw made before. There was a try-finally deleting the file, but that doesn't delete the file when the user hits CTRL+C while deleteOnExit() does. So instead of having 2 ways of deleting it was decided to keep the one that works in more cases. For the debugging we an surround it with an `!useDebugAttach()` perhaps?

Reply:
Ok, how about making a copy of the to-be-deleted-on-exit file if `exitStatus != 0`?

Reply:
Wouldn't that leave a whole bunch of files around on the user's system? Remember that this is not a debug option or anything, it will be used on every execution.

Reply:
I would suggest that the copy has a fixed name (e.g. `native-image-failure.args`) to avoid polluting the system. This file will deleted if it exists before running native-image only be left behind when native-image fails. In addition, all this can be activated only when `--verbose` is supplied. It's only in that scenario that someone may be interested in replaying the low-level native image command.

Reply:
@quintesse are you going to address https://github.com/oracle/graal/pull/2443#discussion_r443523222 or should we close this PR? cc @dougxc

Reply:
Done @olpaw @dougxc